src/components/EditableField/EditableField.types.ts

```diff
@@ -74,21 +94,27 @@ export interface EditableFieldProps {
   onEnter: (args: { name: string; value: FieldValue[]; event: Event }) => void
   onEscape: (args: { name: string; value: FieldValue[]; event: Event }) => void
   onAdd: (args: { name: string; value: FieldValue[] }) => void
-  onCommit: (args: { name: string; value: FieldValue[] }) => void
+  onCommit: (args: {
+    name: string
+    value: FieldValue[]
+    data: CommitData
+  }) => void
   onDelete: (args: { name: string; value: FieldValue[]; event: Event }) => void
   onDiscard: (args: { value: FieldValue[] }) => void
+  validate: (args: { value: string; name: string }) => Promise<Validation>
```

Comment:
@plbabin I'm having some difficult integrating this for network errors. I'm thinking we should either accept validationInfo as props to override the internal state so that we can control the component or removing the validate function in favor of passing in the props, only. What do you think?